### PR TITLE
Typecheck SwAV loss

### PR DESCRIPTION
**Pull request by philippmwirth**
# Typecheck SwAV loss

### CODE REVIEW

File: lightly/loss/swav_loss.py [hunk 1]
```diff
@@ -131,10 +132,10 @@ def subloss(self, z: torch.Tensor, q: torch.Tensor):
 
     def forward(
         self,
-        high_resolution_outputs: List[torch.Tensor],
-        low_resolution_outputs: List[torch.Tensor],
-        queue_outputs: List[torch.Tensor] = None,
-    ):
+        high_resolution_outputs: List[Tensor],
+        low_resolution_outputs: List[Tensor],
+        queue_outputs: Union[List[Tensor], None] = None,
```
Comment (MalteEbner) [inline lightly/loss/swav_loss.py:137]: We can also use the new syntax here, we already use it at other places: https://github.com/lightly-ai/lightly/blob/53d1af4c01d674e78d0ba00d38a8239d1c31b7e6/lightly/models/modules/ijepa_timm.py#L103

File: tests/loss/test_swav_loss.py [hunk 2]
```diff
@@ -36,34 +31,25 @@ def test_forward_pass(self):
             for sinkhorn_iterations in range(3):
```
Comment (MalteEbner) [inline tests/loss/test_swav_loss.py:31]: This should be a parametrised test.

File: tests/loss/test_swav_loss.py [hunk 3]
```diff
@@ -36,34 +31,25 @@ def test_forward_pass(self):
...
-        queue_length = 128
         queue = [torch.eye(128, 32) for i in range(n_high_res)]
 
         for n_low_res in range(6):
             for sinkhorn_iterations in range(3):
                 criterion = SwaVLoss(sinkhorn_iterations=sinkhorn_iterations)
                 low_res = [torch.eye(n, n) for i in range(n_low_res)]
+                loss = criterion(high_res, low_res, queue)
+                # loss should be almost zero for unit matrix
+                assert loss.cpu().numpy() < 0.5
```
Comment (MalteEbner) [inline tests/loss/test_swav_loss.py:50]: This should be a parametrised test.

File: tests/loss/test_swav_loss.py [hunk 4]
```diff
@@ -87,16 +69,12 @@ def test_forward_pass_1d(self):
             for sinkhorn_iterations in range(3):
                 criterion = SwaVLoss(sinkhorn_iterations=sinkhorn_iterations)
                 low_res = [torch.eye(n, 1) for i in range(n_low_res)]
+                loss = criterion(high_res, low_res)
+                # loss should be almost zero for unit matrix
+                assert loss.cpu().numpy() < 0.5
```
Comment (MalteEbner) [inline tests/loss/test_swav_loss.py:74]: This should be a parametrised test.

File: tests/loss/test_swav_loss.py [hunk 5]
```diff
@@ -24,10 +22,7 @@ def test__sinkhorn_gather_distributed_dist_not_available(
             SwaVLoss(sinkhorn_gather_distributed=True)
         mock_is_available.assert_called_once()
 
-
-class TestSwaVLossUnitTest(unittest.TestCase):
-    # Old tests in unittest style, please add new tests to TestSwavLoss using pytest.
-    def test_forward_pass(self):
+    def test_forward_pass(self) -> None:
```
Comment (MalteEbner) [inline tests/loss/test_swav_loss.py:25]: This should be a parametrised test instead of iterating with 2 for-loops. 
Same applies to the other tests.

**Review by MalteEbner — COMMENTED** [review]
I would definitely use the new typing syntax wherever possible.

File: lightly/loss/swav_loss.py (hunk 1 above)
Comment (philippmwirth) [reply lightly/loss/swav_loss.py:137]: I'd like to keep consistent with the rest of the file. IMO it doesn't make sense to mix the styles. We should decide for one way and then pull it through the whole code base.

**Review by MalteEbner — APPROVED** [review]
LGTM
